### ISSUE TITLE
Need to handle iCloud account changes

### ISSUE THREAD

**Issue by johnthuss** [nofelmahmood/CKSIncrementalStore#11]
If the user switches iCloud accounts their local data will no longer match the server's data.  In that case you would need initiate a complete two-way sync of all records stored both locally and on the server.  In iOS 9 there is a notification you can subscribe to so you know when the account change happens, but in iOS 8 I believe the only option is to record the user's iCloud ID and poll it for changes periodically (like at startup?).  Alternatively, providing a button to manually initiate a complete sync might be acceptable.

**Comment by nofelmahmood** [comment]
Yes you are right. But personally I think that can be left to the user of this library to handle. Like at the end of the day it's a store added to your app's NSPersistentStoreCoordinator. In case of this you can remove the store and add a new one and start fresh ?

**Comment by acegreen** [comment]
Ok I looked int the project and its actually fairly straight forward. I was just overwhelmed by the “CKSIncrementalStoreCloudStoreSetupOperation” and the “CKSIncrementalStoreSyncOperation” files but it seems the implementation itself is fairly straight forward. You use the two notifications to fetch your data again. 

Can you explain when you use 

NSNotificationCenter.defaultCenter().addObserver(self, selector: "syncFinished:", name: CKSIncrementalStoreDidStartSyncOperationNotification, object: self.cksIncrementalStore)

I’m guessing we can add this observer so that we can get notified when sync is being performed on one device to prohibit sync on our device for the time being? If so the selector name is misleading. 

I’m still waiting for your go ahead to download the updated code. 

Cheers
Ace

**Comment by nofelmahmood** [comment]
I am actually decoupling the code. It will then be more easy to find bugs and fix them and it looks clean too. You don't need to worry about them. The usage will remain the same.

The selector name was for the demo project. It's your choice, name it whatever you like. These two notifications are posted to notify the start and end of the sync operation.

I know its taking time but I assure it will be worth the wait.
I'l close [this](https://github.com/nofelmahmood/CKSIncrementalStore/issues/5) issue when it's completely upgraded to Swift 2.0. You can subscribe to it to be notified when that happens :)

**Comment by acegreen** [comment]
Decoupling what from what? 

How heavy is your code? And does it adapt to any core data store as long as we setup the correct Coordinator info?

**Comment by nofelmahmood** [comment]
Don't worry man ! The decoupling of code would not have any impact on the user side. It's all to make code look clean, readable and of course testable. 
You'l only have to add the store with CKSIncrementalStore type to your app's NSPersistentStoreCoordinator as usual to use it :)

**Comment by acegreen** [comment]
I'm not worried, just curious. 

So I see its swift 2.0 ready?

**Comment by acegreen** [comment]
Did you try this with Apple Watch? I noticed that CloudKit is not available on WatchOS2

**Comment by nofelmahmood** [comment]
Sorry I didn't because of my busy schedule :(
You can check out the master branch right now btw.

**Comment by acegreen** [comment]
Yea I saw the branch but I’m wondering what to do with about the Apple Watch store. I’m surprised Apple made us use dedicated stores for Apple Watch with no iCloud access.

**Comment by acegreen** [comment]
Are you sure 0.5.2 is swift ready?

I ran it and got 34 error messages and 39 warnings
